### ISSUE TITLE
Where weight Update happens while minimising loss? 

### ISSUE THREAD

**Issue by ghost** [dreamgonfly/cnn-from-scratch#1]
This code works perfectly. But I was wondering where exactly the weight update(W1, W2 and W3) is happpening in the code? Also, 

**trainer = tf.train.GradientDescentOptimizer(learning_rate).minimize(loss)**
the above line does not take any weight variables as parameters. So how does it update weights and where the change in weights is done in the code? 
Please help me with this doubt since I've been struggling with it since long time. 
Thank you in advance for your efforts.
